### PR TITLE
Anatomical map tags #4m8hgy

### PR DESCRIPTION
**Pull request by matthewtgilbride**
# Description

Refactor the behavior of clicking on the anatomical map to add a filter instead of updating the search query.  

## Type of change

Delete those that don't apply.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

1. Open the anatomical map on the Find Data - Datasets tab.
2. Click on Heart
3. The results should be filtered, the Heart filter should be selected.
4. Open the map again, click on Heart again.
5. Nothing should change (make sure the query params in the address bar stay the same)
6. Open the map again, click on "Carotid body" (it's in the right part of the neck, you'll probably need to zoom in.)
7. Carotid body should be added.
8. Rinse and repeat - try to add Carotid body a second time and make sure nothing changes.

Please note that this change is dependent upon updating the new Contentful filter "Other Anatomic Elements" once we get a list from Alan.  If you click on an element that isn't yet a filter value, you're going to get weird results, so we need to make sure that list stays up to date.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by mahamzayed** [comment]
Whenever I click on `Heart` in the flatmap, I get this error:

image

The console says `Error: Could not load image because of The source image could not be decoded.. Please make sure to use a supported image type such as PNG or JPEG. Note that SVGs are not supported.` and `Cannot read property split of undefined`. 

I get the same error when I try to select `Carotoid Body`.

**Comment by matthewtgilbride** [comment]
> A couple of things:
> 
> 1. Your test script works as expected, however, every time I either open the map or zoom in with my mouse I get a bunch of these errors in the console:
> 
> image
> 
> 1. Not sure if this has been implemented yet, but whenever I try clicking on something else on the map (such as Lung), the modal disappears and I don't see the filter for that organ. I have verified that the filter appears when I click on Colon, so I'm not sure if some data is available right now and some isn't.

Both are expected.  For 1, we need the full set of things that can possibly be clicked on in the map so we can add them as filters in Contentful.  For 2, we'll need to address how to handle the fact that we are calling it "lungs" while the scaffold calls it "lung".

File: pages/data/index.vue [hunk 1]
```diff
@@ -568,11 +568,33 @@ export default {
 
     onMapClick: function(label) {
       const { query } = this.$route;
+      const labelKey = label.toLowerCase()
+
+      // short circuit if nothing has changed
+      if (query.tags === labelKey || find(t => t === labelKey, (query.tags || '').split(','))) {
```
Comment (mahamzayed) [inline pages/data/index.vue]: Nitpick but should be formatted to look like the following:

```
 // short circuit if nothing has changed
      if (
        query.tags === labelKey ||
        find(t => t === labelKey, (query.tags || '').split(','))
      ) {
        return
      }
```

File: pages/data/index.vue [hunk 2]
```diff
@@ -568,11 +568,33 @@ export default {
 
     onMapClick: function(label) {
```
Comment (mahamzayed) [inline pages/data/index.vue]: Can you add a JSDoc comment to indicate what this function does?

**Pull request merged**